Home_work_2

a33303 opened the pull request:
Все задания выполнены.

a33303 committed:
Home_work_2

Все задания выполнены.

matkowton commented on hw2/public/hw2_1.php:
```diff
@@ -0,0 +1,25 @@
+        return $a - $b;
+    } elseif ($a < 0 && $b < 0) {
+        return $a + $b;
+    } elseif ($a > 0 && $b <= 0) {
```
Это условие избыточно. У нас всего 3 -варианта. Если два не подошли, остается только третий. Поэтому здесь достаточно просто else

matkowton commented on hw2/public/hw2_2.php:
```diff
@@ -0,0 +1,35 @@
+$a = rand (-10, 10);
+$b = rand (-10, 10);
+
+function mul($a, $b) {
```
Речь шла о математических функциях, т.е. над операциями над числами. Контроль типов на тип данных и возвращаемое значение был бы здесь совсем не лишним

matkowton commented on hw2/public/hw2_3.php:
```diff
@@ -0,0 +1,53 @@
+            $result = div($arg1, $arg2);
+            break;
+    }
+    return $result;
```
Если в $operation передано некорректное значение, то вернется 0, ято не говорит о том, что в функции что-то пошло не так

matkowton commented on hw2/public/hw2_4.php:
```diff
@@ -0,0 +1,21 @@
+    <h1>Title 1</h1>
+    <p>Content</p>
+</body>
+<footer style="text-align: center">&copy; <?= date('Y') ?></footer>
```
Разделение логики и верстки. Корректно было бы рассчитать дату в блоке пхп до шаблона и запомнить в переменную. А в шаблоне выводить уже переменную

matkowton commented on hw2/public/hw2_5.php:
```diff
@@ -0,0 +1,21 @@
+$pow = rand (2, 5);
+
+
+function power($val, $pow){
```
Что если я поставлю $pow = -1,например?

matkowton commented on hw2/public/hw2_6.php:
```diff
@@ -0,0 +1,57 @@
+
+function comparison($value, $arr)
+{
+    if (!is_numeric($value) || !is_array($arr) || $value < 0) {
```
is_numeric и is_array легко заменяются контролем типов в описании функции

matkowton commented on hw2/public/hw2_6.php:
```diff
@@ -0,0 +1,57 @@
+    if (!is_numeric($value) || !is_array($arr) || $value < 0) {
+        return 'неверные данные!';
+    }
+    if (preg_match('/[1][1-9]/', substr($value, -2))) {
```
preg_match - слишком тияжелая функция для такой простой задачи

a33303 reviewed (commented):
Поправлю замечания. 
Спасибо! 
Дам обратную связь!

a33303 committed:
Home_work 3.

Сделаны всё, кроме 9*, 10*.
Уезжаю в командировку 15.10. буду 16.10 вечером. Обеаю досдать :)
